tests.js

```diff
@@ -9,9 +9,50 @@
     return;
   }
 
-  function qpair(template, ...values) {
-    return [template, values];
+  // Set up a RegExp subclass so that we can test subclass creation.
+  function SubRegExp(source, flags) {
+    const re = new RegExp(source, flags);
+    Object.setPrototypeOf(re, SubRegExp.prototype);
+    return re;
   }
+  SubRegExp.prototype = Object.create(RegExp.prototype, {
+    constructor: {
+      value: SubRegExp
+    },
+    toString: {
+      value: function () {
+        return 'SubRegExp:/' + this.source + '/' + this.flags;
+      }
+    }
+  });
+  SubRegExp.make = RegExp.make;
+
+  const test = (
+    function testMaker(ctor, flags, x, ...values) {
+      if ('object' === typeof x && 'raw' in x) {
+        // Produce a test record if called as a string template.
+        const template = x;
+        return {
+          ctor: ctor,
+          flags: flags,
+          template: template,
+          values: values.slice()
+        };
+      } else {
+        var makerCtor = ctor;
+        var makerFlags = flags;
+        const args = [x, ...values];
+        for (var i = 0, n = args.length; i < n; ++i) {
+          const arg = args[i];
+          switch (typeof arg) {
+          case 'function': makerCtor = arg; break;
+          case 'string':   makerFlags = arg; break;
+          }
```

Comment:
rather than squishing multiple statements onto one line, could you use if/else instead of a switch statement? `switch` is prone to bugs, and is more verbose and less maintainable than other options.